### DISCUSSION TITLE
Clarity and explicitness of Artifact Design definition

### DISCUSSION THREAD

**Discussion by gregfowlerphd** [CommonCoreOntology/CommonCoreOntologies#502] [Announcements]
The definition reads:

> A Directive Information Content Entity that is a specification of an object, manifested by an agent, intended to accomplish goals, in a particular environment, using a set of primitive components, satisfying a set of requirements, subject to constraints.

Notice that there are seven clauses following ‘that’. While it’s clear that the first clause (‘is a specification of an object’) is intended to describe the Directive ICE itself, its not entirely clear what the other clauses are intended to describe. Many at least _seem_ intended to describe the object specified by the Directive ICE (including ‘intended to accomplish goals’ and ‘in a particular environment’). It would be nice to formulate the definition in such a way that what each clause is intended to describe is clear and explicit.

(Sidenote: I also find the phrase ‘manifested by an agent’ at least a bit obscure, but maybe that’s just me.)

**Comment by ianknowland** [comment]
It's not just you! I've been thinking about this for a bit as well. Parsing this definition has been an issue so I would definitely support reformatting the definition, although I'm more in favor of stripping it down than explicating those clauses.

I would do away with a few of the clauses right off the bat: 

- "In a particular environment" is unnecessary to spell out, I think. It sounds both broad and specific but doesn't strike me as a useful inclusion. My intuition is that some goals can be achieved with designed artifacts in a non-particular environment as well, and so maybe we don't need to include particular environments in the definition. We could just positively define particular environments in domain-level subclasses as necessary.

- "subject to constraints" seems redundant as "satisfying a set of requirements" already points to relevant constraints, but if there's a good reason to express this then I'm all ears.

As far as "manifested by an agent", it sounds like the intent is to say "created by an agent" or something similar. However, I'd shift the agent portion around in the definition; possibly "intended by an agent to accomplish goals" and avoid the question entirely. At this point, though, we're starting to simply point to a Material Artifact directly so that's something to think about.

I also would argue that it should say "Material Artifact" rather than object, since it doesn't appear that an Artifact Design actually prescribes objects. 

So, if we kept the DNA of the original with these changes, maybe something like:

> A Directive Information Content Entity that prescribes a Material Entity intended by an Agent to accomplish goals by using a set of primitive components that satisfy a set of requirements.

This certainly flows better, and saying it prescribes a Material Entity rather than a Material Artifact means we can spell out the rest without being redundant within this definition... but now I'm wondering if leaving it at

> "A Directive Information Content Entity that prescribes some Material Artifact" 

would be sufficient, given that it is already asserted to prescribe some Material Artifact. A lot of the original definition seems more relevant to the definition of the artifact rather than a Dir. ICE that prescribes the artifact, like you said.

**Reply by cameronmore** [reply]
@ianknowland your proposed definition is much more succinct and clear that the original. I'm even thinking that the end--'set of requirements'--should also be cut.

"A Directive Information Content Entity that prescribes a Material Entity intended by an Agent to accomplish goals by using a set of primitive components."

**Comment by gregfowlerphd** [comment]
@ianknowland: My suggestion to explicate the clauses in the current definition was made under the assumption that all the clauses were necessary. If they're not, I'm perfectly happy with some of them being eliminated, as occurs in both your and @cameronmore's proposed definitions.

As far as those definitions go, I think both are much clearer than the current one and manage to more succinctly capture the central features of an artifact design.
